adam/visualization/positioning.py

```diff
@@ -80,13 +92,33 @@ class AxisAlignedBoundingBox:
     corners at (-1, -1, -1) and (1, 1, 1), giving the box a volume of 2(^3)
     """
 
-    center: torch.Tensor = attrib()  # tensor shape: (3,)
+    center: Parameter = attrib()  # tensor shape: (3,)
     scale: torch.Tensor = attrib()  # tensor shape: (3, 3) - diagonal matrix
     # rotation: torch.Tensor = attrib()
 
     def center_distance_from_point(self, point: torch.Tensor) -> torch.Tensor:
         return torch.dist(self.center, point, 2)
 
+    def distance_from_lowest_corner_to_ground(self) -> torch.Tensor:
+        """
+        Returns distance of z value of minimum (w/r/t Z coordinate) corner of the box from origin's Z value.
+        Returns: (1,) tensor
+
+        """
+        corners = self.get_corners()
+        min_corner_z = torch.min(
```

Comment:
This complex expression needs more comments to be clear to a reader (or pull out sub-expressions with descriptive names)